shared/video-sdk/develop/ensure-channel-quality/project-implementation/unity.mdx

```diff
@@ -0,0 +1,259 @@
+
+* A button to switch video quality.
+
+* A label to display last-mile network quality.
+
+To add these elements to the UI, take the following steps:
+
+    1. Right-click **Sample Scene**, then click **UI** > **Button - TextMeshPro**. A button appears in the **Scene** Canvas.
+
+    2. In **Inspector**, rename **Button** to **videoQuality**, then change the following coordinates:
+
+        * **Pos X** - 350
+        * **Pos Y** - 172
+
+    3. Right-click **Sample Scene**, then click **UI** > **Text - TextMeshPro**. A label appears in the **Scene** Canvas.
+
+    4. In **Inspector**, rename **Text(TMP)** to **networkStatus**, then set the **Pos Y** coordinate to `160`.
+
+### Handle the system logic
+
+1. **Add the required library to access the UI elements**
+
+    In `Assets/Agora-RTC-Plugin/Agora-Unity-RTC-SDK/Code/JoinChannelVideo.cs`, add the following after the last import statement:
+    
+    ```csharp
+    using TMPro;
+    ```
+
+1. **Define variables to manage test state and workflow**
+
+    In `Assets/Agora-RTC-Plugin/Agora-Unity-RTC-SDK/Code/JoinChannelVideo.cs`, add the following declarations to the `JoinChannelVideo` class:
+
+    ```csharp
+    private TMP_Text networkStatus; // For updating the network status.
+    private TMP_Text videoQuality; // To access the video quality button programmatically.
+    private bool highQuality = false; // To switch between high and low video quality.
+    private uint remoteUid = 0; // To save the remote user ID.
+    ```
+
+1. **Update the network status indication in the UI**
+
+    To show the network quality result visually to the user, in `Assets/Agora-RTC-Plugin/Agora-Unity-RTC-SDK/Code/JoinChannelVideo.cs`, add the following to the `JoinChannelVideo` class:
+
+    ```csharp
+    private void updateNetworkStatus(int quality)
+    {  
+        if (quality > 0 && quality < 3) 
+        {
+            networkStatus.text = "Network Quality: Perfect";
+            networkStatus.color = Color.green;
+        }
+        else if (quality <= 4) 
+        {
+            networkStatus.text = "Network Quality: Good";
+            networkStatus.color = Color.yellow;
+        }
+        else if (quality <= 6) 
+        {
+            networkStatus.text = "Network Quality: Poor";
+            networkStatus.color = Color.red;
+        }
+        else 
+        {
+            networkStatus.color = Color.white;
+        }
+    }
+    ```
+
+    To programmatically access the UI elements, add the following at the end of `SetupUI` method:
+
+    ```csharp
+    // Access the video quality button.
+    go = GameObject.Find("videoQuality");
+    // Access the label of the video quality button.
+    videoQuality = go.GetComponentInChildren<TextMeshProUGUI>(true);
+    // Change the video quality button text and font size.
+    videoQuality.text = "Low Video Quality";
+    videoQuality.fontSize = 14;
```

Comment:
Can you remove the font size stuff please.

Reply:
Help me here. Can you tell me that where should i move this fontSize stuff?